src/snyk/common/services/featureFlagService.ts

```diff
@@ -17,4 +17,9 @@ export class FeatureFlagService {
       return false;
     }
   }
+
+  async fetchFeatureFlagWithoutErrorHandling(flagName: string): Promise<boolean> {
+    const ffStatus = await this.commandExecutor.executeCommand<FeatureFlagStatus>(SNYK_FEATURE_FLAG_COMMAND, flagName);
+    return ffStatus?.ok ?? false;
+  }
```

Comment:
What if we provide a fallback values instead so we don't need this extra function?

```ts
  async fetchFeatureFlag(flagName: string, fallbackValue = false): Promise<boolean> {
    try {
      const ffStatus = await this.commandExecutor.executeCommand<FeatureFlagStatus>(
        SNYK_FEATURE_FLAG_COMMAND,
        flagName,
      );
      return ffStatus?.ok ?? false;
    } catch (error) {
      console.warn(`Failed to fetch feature flag ${flagName}: ${error}`);
      return fallbackValue;
    }
  }
```

The we change the `setupFeatureFlags` to handle multiple asynchronous operations concurrently with `allSettled`. I'll add and example in that bit, one sec 🙏

Reply:
Sounds good!